### DISCUSSION TITLE
Interfacing quadratic constraints for conic solver interface

### DISCUSSION THREAD

**Discussion by RudolfReiter** [casadi/casadi#3253] [General]
Dear Joris, dear all,
how can one access quadratic constraints with the conic solver interface? Particularly, I want to formulate an MIQCQP, which is supported by many solvers like gurobi.

Best regards,
Rudolf Reiter

**Comment by jgillis** [comment]
Dear Rudolf;

The conic interface in current form only has semidefiniteness constraints (see https://web.casadi.org/python-api/#qp).
There is a helper function `soc` to convert second-order cone constraints to sdp (see https://github.com/casadi/casadi/blob/main/test/python/conic.py#L1217 ). Something similar can be done for quadratic constraints.
